Cookbook/Technical-Documents/AutomateAUIFlow.md

```diff
@@ -0,0 +1,259 @@
+
+Rather than duplicate this code in both functions we have a *helper* function in *BaseScreen.swift*. This helper does a bit more in that it also verifies the state of the button and wait for the desired state up to a specified timeout.
+
+```swift
+    func typeInto(element: XCUIElement, withText input: String, andState state: ElementState = .exists, waiting timeout: TimeInterval = 10.0) {
+        waitAndTap(element: element, withState: state, waiting: timeout)
+        element.typeText(input)
+    }
+    
+    func waitAndTap(element: XCUIElement, withState state: ElementState = .exists, waiting timeout: TimeInterval = 10.0) {
+        tryWait(for: element, with: state, timeout: timeout)
+        element.tap()
+    }
+```
+
+For pressing the login button we can either use **waitAndTap** or call **tap** directly, which is slightly faster. The helpers functions were intended to stabilise test execution by checking the state of the element before interacting or proceeding with the test, and while this has worked in making the tests much more reliable, it has a performance impact. I would only recommend using them when needed, in this example for pressing the login button I will just use the frameworks API directly.
+
+## API Interface
+We created a class **APIInterface.swift** to contain code relating to creating requests, creating and parsing json's. This class is extended with functionality specific extensions to handle request for a particular area **APIInterface+Appointments.swift**. These classes are written in pure Swift and only use *BabylonCore* with no additional Pods. 
+
+### Finishing the Step Definition
+
+Now that were written out API calls and implemented our screen object we can complete out step definition with the following code.
+
+```swift
+        step("I enter credentials for a \"(.*?)\" user") { (user: String) in
+        	  let email = self.generateRandomEmail()
+            self.registerUser(withEmail: email, forCountry: user)
+            
+            LoginScreen().enterCredentialsFor(email)
+            LoginScreen().tapLogin()            
+        }
+```
+
+On completion the user will be signed in with a new user created at the API level for the specified country.
+
+# Test Scheme
+In Babylon we decided to run the tests using a **Fastlane** lane and a separate scheme. The scheme created was called **BabylonAppUITests** and will need to be selected before attempting to run any UI tests.
+
+# How and when do we run UI tests
+Once a new test has been added and if a new feature file was created also added to **UILanes**  the tests will be run nightly as per our CircleCI yml file. Once a lane has been run the results will be published the ios-build channel for anyone in the team to see, though normally the support engineer and senior Automation Engineer will review the results.
```

Comment:
Maybe we should include how to run the test suite on your local branch before merging the PR? Or how to run the smoke test?

Reply:
I will add a section